File: stack/simplify-path/simplify-path.cs
public class Solution
{
    public string SimplifyPath(string path)
    {
        var stack = new Stack<string>();
        foreach (var part in path.Split('/', StringSplitOptions.RemoveEmptyEntries))
        {
            if (part == ".") continue;
            if (part == "..") { if (stack.Count > 0) stack.Pop(); }
            else stack.Push(part);
        }
        return "/" + string.Join("/", stack.Reverse());
    }
}

File: stack/simplify-path/simplify-path.ts
function simplifyPath(path: string): string {
    const stack: string[] = [];
    for (const part of path.split("/")) {
        if (part === "" || part === ".") continue;
        if (part === "..") { if (stack.length) stack.pop(); }
        else stack.push(part);
    }
    return "/" + stack.join("/");
}
